ginivision/src/androidTest/java/net/gini/android/vision/test/PermissionsHelper.java

```diff
@@ -0,0 +1,43 @@
+package net.gini.android.vision.test;
+
+import static android.support.test.InstrumentationRegistry.getInstrumentation;
+import static android.support.test.InstrumentationRegistry.getTargetContext;
+
+import android.os.Build;
+
+public class PermissionsHelper {
+
+    private static final long GRANT_PERMISSION_PAUSE_DURATION = 500;
+
+    public static void grantExternalStoragePermission() throws InterruptedException {
+        grantExternalStoragePermission(getTargetContext().getPackageName());
+    }
+
+    public static void grantExternalStoragePermission(final String packageName)
+            throws InterruptedException {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            getInstrumentation().getUiAutomation().executeShellCommand(
+                    "pm grant " + packageName
+                            + " android.permission.READ_EXTERNAL_STORAGE");
+            getInstrumentation().getUiAutomation().executeShellCommand(
+                    "pm grant " + packageName
+                            + " android.permission.WRITE_EXTERNAL_STORAGE");
+        }
+        // A delay is needed for the permission
+        Thread.sleep(GRANT_PERMISSION_PAUSE_DURATION);
```

Comment:
We can't be sure actually. I got to this result by trial-and-error and with these delays the tests are reliable.

Nevertheless I think I can add a permission check beforehand and sleep only when permission had to be granted.